src/main/java/dinkplugin/notifiers/LevelNotifier.java

```diff
@@ -110,25 +121,44 @@ public void onGameStateChanged(GameStateChanged gameStateChanged) {
         }
     }
 
-    private void handleLevelUp(String skill, int level, int xp) {
+    private void handleLevelUp(Skill skill, int level, int xp) {
         if (!isEnabled()) return;
 
+        Integer previousXp = currentXp.put(skill, xp);
+        if (previousXp == null) {
+            shouldInit.set(true);
+            return;
+        }
+
+        String skillName = skill.getName();
         int virtualLevel = level < MAX_REAL_LEVEL ? level : getLevel(xp); // avoid log(n) query when not needed
-        Integer previousLevel = currentLevels.put(skill, virtualLevel);
+        Integer previousLevel = currentLevels.put(skillName, virtualLevel);
 
         if (previousLevel == null) {
             shouldInit.set(true);
             return;
         }
 
-        if (virtualLevel < previousLevel) {
+        if (virtualLevel < previousLevel || xp < previousXp) {
             // base skill level should never regress; reset notifier state
             reset();
             return;
         }
 
         // Check normal skill level up
-        checkLevelUp(config.notifyLevel(), skill, previousLevel, virtualLevel);
+        final boolean enabled = config.notifyLevel();
+        checkLevelUp(enabled, skillName, previousLevel, virtualLevel);
+
+        // Check if xp milestone reached
+        int xpInterval = config.xpInterval() * 1_000_000;
+        if (enabled && xpInterval > 0 && level >= MAX_REAL_LEVEL && xp > previousXp) {
+            int remainder = xp % xpInterval;
+            if (remainder == 0 || xp - remainder > previousXp) {
```

Comment:
we could also force notif if at 200m, even if not divisible by the interval

but virtual levels already has a notif for 200m & it'd be dank to fire two notifs (tho we *could* selectively only fire 200m here if virtual levels disabled)

Reply:
I'm ok with forcing 200m notifs. It should be added to the tooltip (and the virtual level should have a tooltip for that too tbh)